src/brsTypes/components/BrsComponent.ts

```diff
@@ -27,6 +27,10 @@ export class BrsComponent {
         );
     }
 
+    protected registerMethod(index: string, method: Callable) {
```

Comment:
Does it make sense to call this `appendMethod` instead? I think is clearer since you need a map of methods first or this wont' work.